### PR TITLE
feat(new-version): expose newversion class in a dedicated package to make it reusable

### PR DESCRIPTION
**Pull request by matthieu-crouzet**
## Proposed change
Expose `NewVersion` class in a dedicated package to make it reusable

## Related issues

- :rocket: Feature #2035

### CODE REVIEW

File: packages/@o3r/git-tools/tsconfig.doc.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "extends": "../../../tsconfig.doc",
+  "exclude": [
```
Comment (kpanot) [inline packages/@o3r/git-tools/tsconfig.doc.json:3]: not needed (the patterns listed does not match any file)

File: packages/@o3r/git-tools/package.json [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    "nx": "nx",
+    "ng": "yarn nx",
+    "build": "yarn nx build git-tools",
+    "postbuild": "patch-package-json-main",
+    "prepare:publish": "prepare-publish ./dist",
+    "prepare:build:builders": "yarn cpy 'schematics/**/*.json' dist/schematics && yarn cpy 'collection.json' dist",
+    "build:builders": "tsc -b tsconfig.builders.json --pretty && yarn generate-cjs-manifest"
+  },
+  "peerDependencies": {
+    "semver": "^7.5.2"
```
Comment (fpaul-1A) [inline packages/@o3r/git-tools/package.json:22]: should this be added with the ng-add ?

File: .yarnrc.yml [hunk 3]
```diff
@@ -17,5 +17,9 @@ packageExtensions:
   "@swc-node/register@^1.9.0":
     dependencies:
       "@swc/types": "*"
+  # Could be removed once https://github.com/actions/toolkit/issues/1684 is fixed
```
Comment (vscaiceanu-1a) [inline .yarnrc.yml:20]: can you add a `TODO`? it will be helpful when we search for leftovers

File: packages/@o3r/git-tools/package.json [hunk 4]
```diff
@@ -0,0 +1,64 @@
+{
+  "name": "@o3r/git-tools",
```
Comment (kpanot) [inline packages/@o3r/git-tools/package.json:2]: The name seems quite generic and not really matching the readme description.
Maybe we can give a name to be more specific and indicate that it expose new version compute functions only.
I am not sure that future git tooling will not bring dependencies and then require their own package as well.

File: packages/@o3r/git-tools/testing/setup-jest.ts
Comment (vscaiceanu-1a) [inline packages/@o3r/git-tools/testing/setup-jest.ts:1]: `import 'jest-preset-angular/setup-jest';`

File: tools/github-actions/new-version/packaged-action/LICENSE.txt [hunk 5]
```diff
@@ -69,35 +69,7 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
...
-WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
-DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
-ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
-(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
-LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
-ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
-SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
+@o3r/git-tools
```
Comment (vscaiceanu-1a) [inline tools/github-actions/new-version/packaged-action/LICENSE.txt:72]: new-version

File: tools/github-actions/new-version/packaged-action/index.js.map
Comment (vscaiceanu-1a) [inline tools/github-actions/new-version/packaged-action/index.js.map:1]: you should repackage as you still have references to the old package name

**Review by vscaiceanu-1a — APPROVED** [review]

**Pull request merged**